### ISSUE TITLE
Auto accept transfers from selected devices

### ISSUE THREAD

**Issue by avaidyam** [grishka/NearDrop#92]
Since `Contacts`-mode and `Self-device`-mode aren't available (due to requiring Google server involvement), it might be a good idea to include an option to set "favorite devices" from which all transfers will be auto-accepted.

---

*Also, I shared this comment in #16, but I'm reproducing it here in an open issue:*

FYI Nearby Share (unlike AirDrop) [does support sharing text between devices](https://9to5google.com/2023/05/23/nearby-share-copy-send-text/) (received text is indeed copied to the clipboard and NOT saved as a file). I think it's still worth adding this as a feature.

EDIT: Actually, it seems like in some scenarios, AirDrop does support sharing raw text -- however, the plain text becomes a filename with extension .textClipping (i.e. "This is a test" → ~/Downloads/This is a test.textClipping)

**Comment by grishka** [comment]
The problem is that I don't know of any reliable ways of identifying devices. Names are bad because they're not unique, MAC addresses are bad because they're medium-specific. All unknown fields in the protocol that I've looked at are different for each transfer from the same device. At this point I'm not even sure it is at all possible to identify devices without talking to Google servers.

**Comment by avaidyam** [comment]
A combination of name + photo might work? It's definitely not unique but perhaps that's the warning text when using this feature?

**Comment by grishka** [comment]
What photo?

**Comment by SohamJa1n** [comment]
I was just thinking of this, you could make it so that when a device asks to share, you could add an option called "trust" in which you can use whatever identifier you get from the device such as name although its not unique and it auto accepts it.
Although the names are unique, this requires the devices to be on the same network so it would be rare that someone has the same device name although someone could rename it for malicious intentions.

**Comment by GiorgioMendieta** [comment]
What about `Secure.ANDROID_ID` as found [here](https://stackoverflow.com/questions/4799394/is-secure-android-id-unique-for-each-device)?

**Comment by grishka** [comment]
> What about Secure.ANDROID_ID as found [here](https://stackoverflow.com/questions/4799394/is-secure-android-id-unique-for-each-device)?

Sure, it's easy when you have access to the Android side. You could as well generate a random string on your app's first run and store it in SharedPreferences. The problem is, I don't have access to the Android side. I have to somehow identify devices only using what they send me.

**Comment by slidingshade** [comment]
> Since `Contacts`-mode and `Self-device`-mode aren't available (due to requiring Google server involvement), it might be a good idea to include an option to set "favorite devices" from which all transfers will be auto-accepted.
> 
> _Also, I shared this comment in [#16](https://github.com/grishka/NearDrop/issues/16), but I'm reproducing it here in an open issue:_
> 
> FYI Nearby Share (unlike AirDrop) [does support sharing text between devices](https://9to5google.com/2023/05/23/nearby-share-copy-send-text/) (received text is indeed copied to the clipboard and NOT saved as a file). I think it's still worth adding this as a feature.
> 
> EDIT: Actually, it seems like in some scenarios, AirDrop does support sharing raw text -- however, the plain text becomes a filename with extension .textClipping (i.e. "This is a test" → ~/Downloads/This is a test.textClipping)

Yeah the clipboard sharing would be a nice feature. Tried to use KDE Connect for that purpose but it's not reliable..
